tools/tpm2_pcrextend.c

```diff
@@ -0,0 +1,140 @@
+//**********************************************************************;
+// Copyright (c) 2017, Intel Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice,
+// this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation
+// and/or other materials provided with the distribution.
+//
+// 3. Neither the name of Intel Corporation nor the names of its contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//**********************************************************************;
+
+#include <stdlib.h>
+
+#include <sapi/tpm20.h>
+
+#include "log.h"
+#include "main.h"
+#include "options.h"
+#include "tpm2_alg_util.h"
+#include "tpm2_util.h"
+
+typedef struct tpm_pcr_extend_ctx tpm_pcr_extend_ctx;
+struct tpm_pcr_extend_ctx {
+    TSS2_SYS_CONTEXT *sapi_context;
+    size_t digest_spec_len;
+    tpm2_pcr_digest_spec *digest_spec;
+};
+
+static bool pcr_extend_one(TSS2_SYS_CONTEXT *sapi_context,
+        TPMI_DH_PCR pcr_index, TPML_DIGEST_VALUES *digests) {
+
+    /* TODO SUPPORT AUTH VALUES HERE */
+    TPMS_AUTH_COMMAND session_data = TPMS_AUTH_COMMAND_INIT(TPM_RS_PW);
```

Comment:
I am not sure if this will ever need a password, or other policy bits to actually extend any PCRs. Ill add a ticket for it.